Bottom: 9f39125c28de1b600ec42928d8aea03f6b0a60e0
Top:    9a8281593a7c99be3f0f9987c846c8a0e540c64f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 19:16:17 +0000

Apply modulo arithmetic when computing angles, so output stays user-friendly (and bounded)


---

diff --git a/src/buffer.c b/src/buffer.c
index c03c5d5..a359076 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -1315,6 +1315,7 @@ FreeRotateElementLowLevel (DataType *Data, ElementType *Element,
   /* XXX: Should parse a unit suffix, e.g. "degrees" */
   parse_numeric_attribute (Element, "PCB::rotation", &rotation);
   rotation += angle;
+  rotation = fmod (rotation, 360.L);
   value = g_strdup_printf ("%f degrees", rotation);
   AttributePutToList (&Element->Attributes, "PCB::rotation", value, true);
   g_free (value);
diff --git a/src/rotate.c b/src/rotate.c
index d477d79..789ff81 100644
--- a/src/rotate.c
+++ b/src/rotate.c
@@ -273,6 +273,7 @@ RotateElementLowLevel (DataType *Data, ElementType *Element,
   /* XXX: Should parse a unit suffix, e.g. "degrees" */
   parse_numeric_attribute (Element, "PCB::rotation", &rotation);
   rotation += Number * 90.;
+  rotation = fmod (rotation, 360.L);
   value = g_strdup_printf ("%f degrees", rotation);
   AttributePutToList (&Element->Attributes, "PCB::rotation", value, true);
   g_free (value);
